src/frogger.cpp

```diff
@@ -77,6 +94,24 @@ Rcpp::List run_base_model(const Rcpp::List data,
                                                        age_groups_fert, proj_years);
   leapfrog::TensorMap2<double> births_sex_prop(REAL(data["births_sex_prop"]), num_genders,
                                                proj_years);
+  leapfrog::TensorMap1<double> incidence_rate(REAL(projection_parameters["incidinput"]), proj_years);
+  leapfrog::TensorMap3<double> incidence_relative_risk_age(REAL(projection_parameters["incrr_age"]),
+                                                           age_groups_pop - hiv_adult_first_age_group,
+                                                           num_genders, proj_years);
+  leapfrog::TensorMap1<double> incidence_relative_risk_sex(REAL(projection_parameters["incrr_sex"]), proj_years);
+  leapfrog::TensorMap3<double> cd4_mortality(REAL(projection_parameters["cd4_mort_full"]),
+                                             disease_stages, age_groups_hiv, num_genders);
+  leapfrog::TensorMap3<double> cd4_progression(REAL(projection_parameters["cd4_prog_full"]),
+                                               disease_stages - 1, age_groups_hiv, num_genders);
+  Rcpp::IntegerVector v = Rcpp::as<Rcpp::IntegerVector>(projection_parameters["artcd4elig_idx"]);
+  leapfrog::Tensor1<int> artcd4elig_idx(proj_years);
+  for (int i = 0; i < proj_years; ++i) {
+    // 0-based indexing in C++ vs 1-based indexing in R
+    artcd4elig_idx(i) = v[i] - 1;
+  }
```

Comment:
all of this can go into a helper that does the base 0 to base 1 conversion and also check the upper bounds

Reply:
Added to TODO list in README